flo.gemspec

```diff
@@ -23,15 +23,15 @@ Gem::Specification.new do |spec|
   spec.test_files    = spec.files.grep(%r{^test/})
   spec.require_paths = ['lib']
 
-  spec.add_dependency 'cleanroom'
-  spec.add_dependency 'gpgme'
-  spec.add_dependency 'cri'
+  spec.add_dependency 'cleanroom', '~> 1.0.0'
+  spec.add_dependency 'gpgme', '~> 2.0.18'
```

Comment:
I would like to support the widest range of dependency versions possible.  Users of Flo should not have to upgrade/downgrade gems they are already using just because we chose arbitrarily narrow versions for our dependencies.

However, this can potentially lead to issues where Flo is not compatible with certain versions of the declared dependencies.  Perhaps we can utilize https://github.com/thoughtbot/appraisal or something similar so we can determine if there are any versions that are not compatible.

Reply:
Understood.

A new minor release of cri broke things, so to avoid that from happening again in future, I thought we'd pessimistically lock it and the other gem dependencies to the teeny version (so from `2.15.3` now, we would allow `2.15.4 - 2.15.9` in future but not `2.16`).

For cri I think it comes down to if we think its maintainers broke semantic versioning or not. If their move to freezing their internal structures that caused the issue is something that should have gone into a major version change, then we shouldn't trust future upgrades and we should lock as pessimistically as possible. If the change is seen as reasonable for a minor version bump and our breakage is just seen as an edge case that was unreasonable for cri to expect, then we can go on trusting cri with less (or no) pessimism.

Let's discuss offline and I'll open a new PR with whatever we decide. I see a few options:

• Just lock cri, not the other gems - maybe less pessimistically
• Lock everything, but support minor upgrades instead of just teeny ones ( `'gpgme', '~> 2.0'` instead of `'gpgme' '~> 2.0.18'` )
• Don't lock anything

I think it's usually a good pattern to do at least major version locking for first party gems that we bring in directly (and don't pick up along the way from bringing in others), but Flo's objective of fitting in with lots of projects and environments with their own existing setups could be an entirely valid exception to the rule of that practice.

Reply:
Removed all locks.